### PR TITLE
Fix #3317: Update old todos

### LINKED ISSUE #3317: Update all the TODOs of the repository that are not updated correctly
We want to ensure that all the TODOs correspond to open issues on Github. Hence, all those issues should be reopened which are closed, and whose TODOs are still present in the codebase. If the TODOs are no longer needed, then we should remove/update them. Also, we should ensure that all the TODOs are correctly formatted as per the format: TODO(#ISSUE_NUMBER): <todo_description>

**Target PR date**: 2021-07-20
**Target merge date**: 2021-07-23

### PR DESCRIPTION
**Pull request by Sparsh1212**
## Explanation

Fixes #3317: Correct poorly formatted todos and update those which doesn't correspond to an open issue.

Issues that we should reopen because they aren't resolved, or issues which have additional work to complete:
- [x] [Paused activity does not get LiveData update onResume #322](https://github.com/oppia/oppia-android/issues/322)
- We need to audit the TODO cases to make sure they're resolved (though they might be per Viktoriia's comment in which case we should remove the TODO)
- [x] [Add NetworkModule to ApplicationModule and tests once migrated off of Moshi #1675](https://github.com/oppia/oppia-android/issues/1675)
- One test still needs to be updated (per the TODO) to finish resolving this; we should just fix that and remove the TODO rather than reopening the issue

Issues whose TODOs need to be repurposed to new/other issues:
- [x] [Add Bazel Linter to the project #1736](https://github.com/oppia/oppia-android/issues/1736)
  - New issue should be filed
- [x] [Addition of Work Manager [GSoC'20][Milestone 3]  #1106](https://github.com/oppia/oppia-android/issues/1106)
  - New issue should be filed
- [x] [Implement TopicController #15](https://github.com/oppia/oppia-android/issues/15)
  - New issue should be filed
- [x] [Introduce thread-safe MediatorLiveData or alternative #90](https://github.com/oppia/oppia-android/issues/90)
  - New issue should be filed
- [x] [Add temporary offline support for audio & image streaming #386](https://github.com/oppia/oppia-android/issues/386)
  - Use issue #169 instead of #386 for this TODO
- [x] [Generic data provider infrastructure #6](https://github.com/oppia/oppia-android/issues/6)
  - Except for TODO in DataProvider.kt; that one can be removed
- [x] [Migrate custom recycler view adapters over to the generic bindable adapter #216](https://github.com/oppia/oppia-android/issues/216)
  - Should be updated to reference #632 instead
- [x] [Full UI: ItemSelectionInteractionView [Blocked: #15, #154] #32](https://github.com/oppia/oppia-android/issues/32)
  - New issue should be filed
- [x] [Lowfi Implementation of My Downloads #1806](https://github.com/oppia/oppia-android/issues/1806)
  - Work with Akshay to find/file replacement issues for each existing TODO
- [x] [Introduce dependency injection support #74](https://github.com/oppia/oppia-android/issues/74)
  - New issue should be filed
- [x] [Create backend storage models for feedback reporting #2801](https://github.com/oppia/oppia-android/issues/2801)

Issues whose TODOs are also resolved and thus should be removed from code, or TODOs which do not need to be resolved:
- [x] [Add tests for score and mastery calculations #2738](https://github.com/oppia/oppia-android/issues/2738)
- [x] [Implement TopicController #15](https://github.com/oppia/oppia-android/issues/15)
- [x] [Add support for refresher explorations in the exploration player #182](https://github.com/oppia/oppia-android/issues/182)
- [x] [Implement StoryProgressController #21](https://github.com/oppia/oppia-android/issues/21)
- [x] #1104: Remove TODO and add KDoc

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ ] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by vinitamurthi — APPROVED** [review]
LGTM for codeowner files!

**Review by anandwana001 — APPROVED** [review]
Code owner files LGTM

**Review by Sarthak2601 — APPROVED** [review]
LGTM for #1104's to-do change.

**Review by rt4914 — APPROVED** [review]
Approved for code-owner files. Thanks

**Comment by Sparsh1212** [comment]
@BenHenning I and @prayutsu did some research on the TODO for issue #322, but we weren't able to reach a conclusion. It looks like it needs a deeper investigation. Repurposed the current TODO to a new issue so that this doesn't block this PR.

File: app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityProdTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityProdTest.kt:356]: @anandwana001 will these actually be addressed with your changes?

File: data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt
Comment (BenHenning) [inline data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt]: @jcqli won't this be addressed by your pending PR?

**Review by BenHenning — COMMENTED** [review]
Thanks @Sparsh1212. Generally LGTM, just looking to get confirmation from two other team members before approving these updates.

It's great to see the TODOs becoming up-to-date & correctly formatted. :)

File: data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt
Comment (Sparsh1212) [reply data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt]: @BenHenning Just adding this thread (https://github.com/oppia/oppia-android/pull/3508#issuecomment-890165282) as we have received confirmation from @jcqli.
I have created a new issue according to this conversation i.e #3636.

File: app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityProdTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityProdTest.kt:356]: No, we need these todo, with my ui driven thing, there will be no my downloads menu.

File: data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt
Comment (BenHenning) [reply data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt]: Sorry to be clear: I'm fairly certain this is being addressed by #2878 & https://github.com/oppia/oppia/pull/12367. Rather than introducing a new issue, why not just use #3016 since that's already being tracked?

Comment (Sparsh1212) [reply data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt]: @jcqli Can you confirm if I should update this TODO to use issue no? #3016 and keep the description the same as before or shall I go with the present iteration?

**Comment by Sparsh1212** [comment]
@BenHenning 
We probably missed the TODO for issue #114.
https://github.com/oppia/oppia-android/blob/HEAD/domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt#L106

File: data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt
Comment (jcqli) [reply data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt]: Ah I see what you mean -- I originally didn't want #2878 to be blocked by oppia/oppia#12367 but I think that will be fine. Let's use #3016 and it will be fixed in #2878.

Comment (Sparsh1212) [reply data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt]: Done. Thanks.

**Review by jcqli — APPROVED** [review]
LGTM for issue #3016! Thanks for your patience

**Comment by BenHenning** [comment]
@Sparsh1212 let's file a new issue to track the work for #114's TODO.

File: data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt
Comment (BenHenning) [reply data/src/main/java/org/oppia/android/data/backends/gae/model/GaeFeedbackReport.kt]: Thanks @jcqli!

**Comment by Sparsh1212** [comment]
@BenHenning Filed new issue #3646 as a replacement for #114. PTAL.

**Review by BenHenning — APPROVED** [review]
Thanks @Sparsh1212! LGTM.

**Comment by BenHenning** [comment]
Merging since the Bazel tests should be fine, and they seem to be hanging at the moment.

**Pull request merged**
